File: row-change-event-channel/src/test/java/nl/haploid/event/channel/TestData.java
package nl.haploid.event.channel;

import nl.haploid.event.AtomChangeEvent;
import nl.haploid.event.channel.repository.AtomChangeEventDmo;

import java.util.Random;
import java.util.UUID;

public class TestData {

	public static final String ATOM_LOCUS = "sunnydale";

	public static final String ATOM_TYPE = "spike";

	public static AtomChangeEvent atomChangeEvent() {
		final AtomChangeEvent event = new AtomChangeEvent();
		event.setAtomId(nextLong());
		event.setAtomLocus(ATOM_LOCUS);
		event.setAtomType(ATOM_TYPE);
		return event;
	}

	public static long nextLong() {
		return new Random().nextLong();
	}

	public static AtomChangeEventDmo atomChangeEventDmo() {
		final AtomChangeEventDmo event = new AtomChangeEventDmo();
		event.setAtomId(nextLong());
		event.setAtomLocus(ATOM_LOCUS);
		event.setAtomType(ATOM_TYPE);
		return event;
	}

	public static String topic() {
		return UUID.randomUUID().toString();
	}
}

File: event-emitter/src/test/scala/nl/haploid/octowight/emitter/TestData.scala
package nl.haploid.octowight.emitter

import java.util.{Random, UUID}

import nl.haploid.octowight.AtomChangeEvent

object TestData {

  val AtomCategory = nextString

  val AtomOrigin = nextString

  def atomChangeEvent = AtomChangeEvent(id = nextLong, origin = AtomOrigin, category = AtomCategory)

  def nextLong: Long = new Random().nextLong

  def nextString = UUID.randomUUID.toString
}
